src/storage-client/src/source/persist_source.rs

```diff
@@ -331,36 +351,60 @@ impl PersistSourceDataStats for PersistSourceDataStatsImpl<'_> {
         num_oks.map(|num_oks| num_results - num_oks)
     }
 
-    fn col_min<'a>(&'a self, idx: usize, _arena: &'a RowArena) -> Option<Datum<'a>> {
-        // The `RowArena` is unused now, but it will be needed for things like
-        // e.g. `Datum::List`.
+    fn col_min<'a>(&'a self, idx: usize, arena: &'a RowArena) -> Option<Datum<'a>> {
+        struct ColMin<'a>(&'a dyn DynStats, &'a RowArena);
+        impl<'a> DatumToPersistFn<Option<Datum<'a>>> for ColMin<'a> {
+            fn call<T: DatumToPersist>(self) -> Option<Datum<'a>> {
+                let ColMin(stats, arena) = self;
+                let stats = downcast_stats::<T::Data>(stats)?;
+                let mut buf = Row::default();
```

Comment:
`make_datum` does ~exactly this under the hood, in case that feels tidier.

Reply:
nice TIL! done